### ISSUE TITLE
Identifiers with ' (single quotes) parse but generate illegal C

### ISSUE THREAD

**Issue by gstew5** [dimitriv/Ziria#109]
The following code:
```
fun foo'(x : int) { return x }
let comp main = read[int] >>> map foo' >>> write[int]
```
generates illegal C (C identifiers cannot contain single quote), causing `gcc` warnings and errors:
```
gcc -std=c99 -msse3 -I . -c test.c -o test.o
test.c:26:10: warning: missing terminating ' character [enabled by default]
 int32 foo'_ln1_1(int32 x2);
          ^
...
```

Two potential fixes:
* Catch single-quote ids at parse-time (by making them illegal), or
* Choose related (fresh) names for these identifiers before code generation.

**Comment by siddhanathan** [comment]
Possible duplicate: https://github.com/dimitriv/Ziria/issues/80

**Comment by gstew5** [comment]
Closed as duplicate of #80
